File: examples/ruby/send_sync.rb
# frozen_string_literal: true

# Run with: WALINKO_API_KEY=walk_live_... ruby examples/ruby/send_sync.rb
#
# This example demonstrates the *target* API for Walinko::Client#messages#send.
# It will start working when the Ruby SDK reaches 0.1.0.

require 'walinko'

client = Walinko::Client.new(api_key: ENV.fetch('WALINKO_API_KEY'))

result = client.messages.send(
  device_id:     1,
  template_id:   12,
  variant_index: 0,
  phone:         '+8801617738431',
  variables:     { name: 'Kazi', dist: 'Dhaka' }
)

puts "tracking_id:   #{result.tracking_id}"
puts "wa_message_id: #{result.wa_message_id}"
puts "status:        #{result.status}"

File: examples/php/send_sync.php
<?php

declare(strict_types=1);

// Run with: WALINKO_API_KEY=walk_live_... php examples/php/send_sync.php
//
// Synchronous send: blocks until the WhatsApp gateway acknowledges
// delivery (or the server's 15s timeout fires).

require __DIR__ . '/../../sdks/php/vendor/autoload.php';

use Walinko\Client;

$apiKey = getenv('WALINKO_API_KEY');
if ($apiKey === false || $apiKey === '') {
    fwrite(\STDERR, "Set WALINKO_API_KEY first.\n");
    exit(1);
}

$client = new Client(['api_key' => $apiKey]);

$result = $client->messages->send([
    'device_id'     => 1,
    'template_id'   => 12,
    'variant_index' => 0,
    'phone'         => '+8801617738431',
    'variables'     => ['name' => 'Kazi', 'dist' => 'Dhaka'],
]);

echo "tracking_id:   {$result->trackingId}\n";
echo "wa_message_id: {$result->waMessageId}\n";
echo "status:        {$result->status}\n";
